### ISSUE TITLE
Support beyond purs 0.12 ?

### ISSUE THREAD

**Issue by p2327** [nwolverson/vscode-ide-purescript#140]
Is there any plans to support new versions of Purescript? At the moment I am getting errors using Purescript 0.13.6

Thanks!

**Comment by kritzcreek** [comment]
@p2327 The plugin works fine with 0.13.6. What's the particular error you're getting?

**Comment by p2327** [comment]
@kritzcreek hey there I am getting this error message
```
It doesn't look like the workspace root is a PureScript project (has bower.json/psc-package.json/spago.dhall). The PureScript project should be opened as a root workspace folder.
```

**Comment by kritzcreek** [comment]
You should open VSCode in the root of the PureScript project, as that error says. It's got nothing to do with versions. If it's nested within another project you've got two options:

1. Contribute a feature allowing the VSCode plugin to run a language server in a non-root directory
2. Use multiple VSCode instances with different root directories

**Comment by nwolverson** [comment]
What @kritzcreek says is not wrong, but actually VSCode also has a concept of "workspace folders", where essentially a workspace has multiple root folders - so long as the purescript project is one of those root folders, you should be good.

**Comment by nwolverson** [comment]
Hi @p2327, did this resolve the issue for you?

**Comment by p2327** [comment]
@nwolverson that sounds like it should od the trick however I dont know how to set multiple root folders in a workspace

Im basically opening my ```../projects``` as workspace and then have various project folders there. The Purescript code I am working on is inside a python project folder, so ```../projects/pythonprojects/purescriptfolder```

**Comment by nwolverson** [comment]
@p2327 You can use the command "Add folder to workspace" from the command palette or from right clicking in the file pane on the left

https://code.visualstudio.com/docs/editor/multi-root-workspaces

**Comment by jamesdbrock** [comment]
@nwolverson said:

>  VSCode also has a concept of "workspace folders", where essentially a workspace has multiple root folders - so long as the purescript project is one of those root folders, you should be good.

That sounds right. 

https://code.visualstudio.com/api/references/vscode-api#workspace

And it looks like this extension has been written to support `workspaceFolders` instead of `rootPath`:

https://github.com/nwolverson/vscode-ide-purescript/blob/80cc5fad918f0a31e78005078f8206a61d55aecb/src/extension.ts#L66-L77

But I'm having trouble making this work.... I have a `spago`-based project in a subdirectory of a `.code-workspace`. I have added the subdirectory “Folder” to the `.code-workspace`, and configured a `spago` __Purescript: Build Command__ in the “Folder” `subdirectory/.vscode/settings.json`. But when I open a `.purs` file in the subdirectory, it seems that my `spago` build command runs with the wrong `PWD` and I get a *vscode* error __Problem running build: didn't find JSON output__

I'm still investigating. Anyway, thank you for this extension @nwolverson , in general it works really well.

**Comment by jamesdbrock** [comment]
Similar issue https://github.com/nwolverson/vscode-ide-purescript/issues/124

**Comment by jamesdbrock** [comment]
Creating a separate `.code-workspace` in the `spago`-based project subdirectory works great.

But workspace folders for a `.code-workspace` in a parent directory doesn't seem to correctly run its commands with a `PWD` in the subdirectory where the `spago.dhall` file is.

**Comment by nwolverson** [comment]
@jamesdbrock I just tested this scenario, seems to work perfectly here.

```
[workspace-test]$ ls
project-1  project-2  project.code-workspace
```

```
{
	"folders": [
		{
			"path": "project-1"
		},
		{
			"path": "project-2"
		}
	],
	"settings": {}
}
```

A folder-level setting for `purescript.buildCommand` is picked up, build runs correctly.

What makes you think the PWD is incorrect? Can you attach logging from the Output pane.

I had a vague inkling that there may be an issue if a file is present in multiple workspace folders - eg one workspace folder is contained within another in the FS - but I can't reproduce that just now. In that case it can be worth reordering the folders so that the nested folder is on top.

**Comment by jamesdbrock** [comment]
Correction:  workspace folders for a `.code-workspace` in a parent directory works great, and correctly runs its commands with a `PWD` in the subdirectory where the `spago.dhall` file is.

My problem is that I was trying to customize the `output` directory like this:

```json
"purescript.buildCommand": "spago build --purs-args '--json-errors --output .psc-ide-output'",
"purescript.outputDirectory": ".psc-ide-output/",
```

**Comment by jamesdbrock** [comment]
> I just tested this scenario, seems to work perfectly here.

Yes, it works. Thanks for testing. I was wrong about why my configuration was failing.
